src/lib/bootstrap_controller/bootstrap_controller.ml

```diff
@@ -243,8 +238,46 @@ end = struct
           @@ on_transition t ~sender ~root_sync_ledger transition
         else Deferred.unit )
 
+  let download_best_tip ~network ~logger ~trust_system ~verifier
+      ({With_hash.data= initial_root_transition; _}, _) =
+    let num_peers = 8 in
+    let peers = Network.random_peers network num_peers in
+    Logger.info logger
+      !"Number of peers that we are sampling: %i"
+      (List.length peers) ~module_:__MODULE__ ~location:__LOC__ ;
+    Deferred.Or_error.find_map_ok peers ~f:(fun peer ->
+        let open Deferred.Or_error.Let_syntax in
+        let initial_consensus_state =
+          External_transition.consensus_state initial_root_transition
+        in
+        let%bind peer_best_tip =
+          Network.get_bootstrappable_best_tip network peer
+            initial_consensus_state
+        in
+        let open Deferred.Let_syntax in
+        match%bind
+          Sync_handler.Bootstrappable_best_tip.verify ~logger ~verifier
+            initial_consensus_state peer_best_tip
+        with
+        | Ok {Coda_intf.Best_tip_verification_result.best_tip; root} ->
+            Deferred.Or_error.return (best_tip, root, peer)
```

Comment:
Please use a record or a tag this tuple with anonymous variants. This anonymous tuple makes it very hard to read the pattern you are matching against when you call `download_best_tip`.